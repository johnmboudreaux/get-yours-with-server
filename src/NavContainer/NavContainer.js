import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

class NavContainer extends React.Component {

  constructor(props, context) {
    super(props, context);
  }

  render() {
    return (
      <aside className="main-aside">
        <nav className="nav">
          <ul className="link-container">
            <li>
              <Link to='/'>Personal Recomendations</Link>
            </li>
            <li>
              <Link to='/Baby'>Baby</Link>
            </li>
            <li>
              <Link to='/Camping'>Camping</Link>
            </li>
          </ul>
        </nav>
      </aside>
    );
  }

}

NavContainer.propTypes = {};

export default NavContainer;
