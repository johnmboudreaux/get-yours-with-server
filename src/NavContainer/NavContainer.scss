.link-container {
  margin: 0;
  padding: 0;
  color: #02a9ea;
}

.main-aside {
  background-color: #171b1e;
}

.logo {
  border-bottom: 1px solid #333333;
  color: #02a9ea;
  font-size: 40px;
  font-weight: bold;
  padding: 30px 0 30px 30px;
}

.nav-links {
  border-bottom: 1px solid #333333;
  color: #02a9ea;
  font: bold 11px Arial;
  font-size: 16px;
  padding: 20px 70px 25px 35px;
  list-style: none;
}

ul.link-container li.nav-links a {
  color: #02a9ea;
}
