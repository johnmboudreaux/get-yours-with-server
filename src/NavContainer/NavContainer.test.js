import configureStore from 'redux-mock-store';
import { shallow, mount } from 'enzyme';
import NavContainer from './NavContainer';
import React from 'react';
import { MemoryRouter } from 'react-router';

const middleware = [];
const mockStore = configureStore(middleware);
const initialState = false;
const store = mockStore(initialState);

describe('NavContainer snapshot', () => {

  it('should always match the snapshot', () => {
    const wrapper = shallow(<NavContainer store={store} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('toggleMenu should update state', () => {
    const search = mount(<NavContainer />);

    const button = search.find('button');

    expect(search.state().menuShowing).toEqual(false);

    button.simulate('click');

    expect(search.state().menuShowing).toEqual(true);
  });

});
