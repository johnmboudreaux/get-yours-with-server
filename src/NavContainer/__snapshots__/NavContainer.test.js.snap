// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavContainer snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavContainer
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo"
        >
          Get Yours
        </div>,
        <button
          className="show-menu-button"
          onClick={[Function]}
        >
          Show Menu
        </button>,
        <nav
          className="nav"
        >
          <ul
            className="link-container"
          >
            <li
              className="nav-links"
            >
              <Link
                replace={false}
                to="/"
              >
                Personal Recomendations
              </Link>
            </li>
            <li
              className="nav-links"
            >
              <Link
                replace={false}
                to="/Baby"
              >
                Baby
              </Link>
            </li>
            <li
              className="nav-links"
            >
              <Link
                replace={false}
                to="/Camping"
              >
                Camping
              </Link>
            </li>
          </ul>
        </nav>,
      ],
      "className": "main-aside ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Get Yours",
          "className": "logo",
        },
        "ref": null,
        "rendered": "Get Yours",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Show Menu",
          "className": "show-menu-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Show Menu",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="link-container"
          >
            <li
              className="nav-links"
            >
              <Link
                replace={false}
                to="/"
              >
                Personal Recomendations
              </Link>
            </li>
            <li
              className="nav-links"
            >
              <Link
                replace={false}
                to="/Baby"
              >
                Baby
              </Link>
            </li>
            <li
              className="nav-links"
            >
              <Link
                replace={false}
                to="/Camping"
              >
                Camping
              </Link>
            </li>
          </ul>,
          "className": "nav",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="nav-links"
              >
                <Link
                  replace={false}
                  to="/"
                >
                  Personal Recomendations
                </Link>
              </li>,
              <li
                className="nav-links"
              >
                <Link
                  replace={false}
                  to="/Baby"
                >
                  Baby
                </Link>
              </li>,
              <li
                className="nav-links"
              >
                <Link
                  replace={false}
                  to="/Camping"
                >
                  Camping
                </Link>
              </li>,
            ],
            "className": "link-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/"
                >
                  Personal Recomendations
                </Link>,
                "className": "nav-links",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "Personal Recomendations",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Personal Recomendations",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/Baby"
                >
                  Baby
                </Link>,
                "className": "nav-links",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "Baby",
                  "replace": false,
                  "to": "/Baby",
                },
                "ref": null,
                "rendered": "Baby",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/Camping"
                >
                  Camping
                </Link>,
                "className": "nav-links",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "Camping",
                  "replace": false,
                  "to": "/Camping",
                },
                "ref": null,
                "rendered": "Camping",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "nav",
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo"
          >
            Get Yours
          </div>,
          <button
            className="show-menu-button"
            onClick={[Function]}
          >
            Show Menu
          </button>,
          <nav
            className="nav"
          >
            <ul
              className="link-container"
            >
              <li
                className="nav-links"
              >
                <Link
                  replace={false}
                  to="/"
                >
                  Personal Recomendations
                </Link>
              </li>
              <li
                className="nav-links"
              >
                <Link
                  replace={false}
                  to="/Baby"
                >
                  Baby
                </Link>
              </li>
              <li
                className="nav-links"
              >
                <Link
                  replace={false}
                  to="/Camping"
                >
                  Camping
                </Link>
              </li>
            </ul>
          </nav>,
        ],
        "className": "main-aside ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Get Yours",
            "className": "logo",
          },
          "ref": null,
          "rendered": "Get Yours",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Show Menu",
            "className": "show-menu-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Show Menu",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="link-container"
            >
              <li
                className="nav-links"
              >
                <Link
                  replace={false}
                  to="/"
                >
                  Personal Recomendations
                </Link>
              </li>
              <li
                className="nav-links"
              >
                <Link
                  replace={false}
                  to="/Baby"
                >
                  Baby
                </Link>
              </li>
              <li
                className="nav-links"
              >
                <Link
                  replace={false}
                  to="/Camping"
                >
                  Camping
                </Link>
              </li>
            </ul>,
            "className": "nav",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="nav-links"
                >
                  <Link
                    replace={false}
                    to="/"
                  >
                    Personal Recomendations
                  </Link>
                </li>,
                <li
                  className="nav-links"
                >
                  <Link
                    replace={false}
                    to="/Baby"
                  >
                    Baby
                  </Link>
                </li>,
                <li
                  className="nav-links"
                >
                  <Link
                    replace={false}
                    to="/Camping"
                  >
                    Camping
                  </Link>
                </li>,
              ],
              "className": "link-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/"
                  >
                    Personal Recomendations
                  </Link>,
                  "className": "nav-links",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Personal Recomendations",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Personal Recomendations",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/Baby"
                  >
                    Baby
                  </Link>,
                  "className": "nav-links",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Baby",
                    "replace": false,
                    "to": "/Baby",
                  },
                  "ref": null,
                  "rendered": "Baby",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/Camping"
                  >
                    Camping
                  </Link>,
                  "className": "nav-links",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Camping",
                    "replace": false,
                    "to": "/Camping",
                  },
                  "ref": null,
                  "rendered": "Camping",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "nav",
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
